rspace/src/main/scala/coop/rchain/rspace/pure/StoreInstances.scala

```diff
@@ -0,0 +1,411 @@
+package coop.rchain.rspace.pure
+
+import java.nio.ByteBuffer
+
+import cats.data.{ReaderT}
+import coop.rchain.catscontrib.Capture
+import coop.rchain.rspace.Serialize
+import coop.rchain.rspace.internal._
+import org.lmdbjava.{CursorIterator, KeyRange, Txn}
+import java.security.MessageDigest
+
+import cats.Monad
+import coop.rchain.rspace.util.{ignore, withResource}
+import coop.rchain.rspace.internal.scodecs._
+import scodec.Codec
+import scodec.bits.{BitVector, ByteVector}
+
+import scala.collection.immutable.Seq
+import coop.rchain.rspace.util
+
+import scala.collection.JavaConverters._
+
+object StoreInstances {
+
+  implicit def storeLMDB[F[_]: Monad, C, P, A, K](
+      implicit
+      serializeC: Serialize[C],
+      serializeP: Serialize[P],
+      serializeA: Serialize[A],
+      serializeK: Serialize[K],
+      captureF: Capture[F]): Store[ReaderT[F, LMDBContext, ?], C, P, A, K] =
+    new Store[ReaderT[F, LMDBContext, ?], C, P, A, K] {
+
+      private[this] def capture[X](x: X): F[X] = captureF.capture(x)
+
+      type T = Txn[ByteBuffer]
+
+      def createTxnRead(): ReaderT[F, LMDBContext, T] =
+        ReaderT(ctx => capture(ctx.env.txnRead))
+
+      def createTxnWrite(): ReaderT[F, LMDBContext, T] =
+        ReaderT(ctx => capture(ctx.env.txnWrite))
+
+      def withTxn[R](txn: T)(f: T => R): R =
+        try {
+          val ret: R = f(txn)
+          txn.commit()
+          ret
+        } catch {
+          case ex: Throwable =>
+            txn.abort()
+            throw ex
+        } finally {
+          txn.close()
+        }
+
+      type H = ByteBuffer
+
+      private[this] def hashChannelsInner(channels: Seq[C]): ByteBuffer =
+        hashBytes(toByteBuffer(channels))
+
+      override def hashChannels(channels: Seq[C]): ReaderT[F, LMDBContext, ByteBuffer] =
+        ReaderT.pure(hashChannelsInner(channels))
+
+      override def getChannels(txn: T, channelsHash: H): ReaderT[F, LMDBContext, Seq[C]] =
+        ReaderT(ctx =>
```

Comment:
In terms of formatting & braces, I like doing:

```scala
        ReaderT { ctx =>
          capture {
            Option(ctx.dbKeys.get(txn, channelsHash))
              .map(fromByteBuffer[C])
              .getOrElse(Seq.empty[C])
          }
        }
```

I think it "breathes" a bit better.